### ISSUE TITLE
Enhancement: multiple offset in mixes

### ISSUE THREAD

**Issue by Joetrin** [FrSkyRC/ETHOS-Feedback-Community#378]
Would it be possible to put Offset in the Mix, please.

It is used to do idle up or high idle on gas engines and also used for Crow setup for airplanes.  It can also be used for flaperons and many other uses.

Thank you

**Comment by hrenz** [comment]
Yes that is important
Weight and Offset always belong together
to be able to adjust a range.
Only weight is not enough

**Comment by Hobby4life** [comment]
I must agree, I have now programmed more then 15 models, last few days.. and everytime in needed offset, I ended up using a curve. I gues there should be no need for that.
Also to mention then for every mix SLOW and DELAY must be involved.. not only on outputs.

**Comment by aviat40** [comment]
+1

**Comment by bsongis-frsky** [comment]
For some mixes it makes no sense to me, unless your centering is not good, but it will result in a very bad result.
On which mixes do you need this offset @Joetrin ?

**Comment by Hobby4life** [comment]
It is nice that Ethos provides us a pre made mixer I love it, but in some cases we need offset, instead of setting up a mixer.
We would be free to choose wheter we use offset or not, most of the time I keep getting back to a free mix again.

Same for slow and delay, these always should be available in a mixer, not in outputs.
I sometimes use my flaps to collaborate with my ailerons.. if I set slow in outputs for flaps only, slow is still active while using aileron+flap control mix flightmode.

In short, Mix does what it says, Every control of a mix must be in that specific mix. at least we clearly **all** expect that it should be there ;-)

**Comment by Liverboy77** [comment]
You would need the offset in flaps to elevator mix.
Example elevator weight is 10% with no offset the elevator will be -10%, 0%, +10%. With offset elevator will be 0%, +10%, +20%.
You need offset other wise with no mix active elevator would be -10% which is not good. I know you can create a curve, but it would be so much faster and easier if there was an offset option.
Thanks

**Comment by Joetrin** [comment]
> For some mixes it makes no sense to me, unless your centering is not good, but it will result in a very bad result.
> On which mixes do you need this offset @Joetrin ?

Throttle - to be able to setup low , mid , high,  idle without changing curves.  Some glow / gas flyers like to setup landing mode (low idle), sport flying ( high idle) 

Ailerons / flaps: allows for quick setup of Crow, flaperons (inboard flaps)

Elevator: makes it simple to setup down elevator offset when flaps are deployed

Hope this helps

**Comment by jfint** [comment]
So far I have needed it in Camber, but it would be separate for flaps and ailerons.  Basically any aircraft location where the sever/motor is not intended to have its normal flight condition at 1500us

**Comment by bsongis-frsky** [comment]
@jfint why don't you use the channel offset?

**Comment by bsongis-frsky** [comment]
@Joetrin same question for Aileron, Flap and Elevator (for Throttle I need to check myself)

**Comment by bsongis-frsky** [comment]
@Liverboy77 how do you use the flaps? on a 3-pos switch, and the "no flaps" position is TOP or BOTTOM?

**Comment by jfint** [comment]
Because there are 4 channels at play there, and if you offset at that oevel it takes away your throw to the point where the mix is useless.

**Comment by jfint** [comment]
Here is a first cut at an F3F model in which I had to use a curve on the flaps mix to offset those surfaces, and then a free mix in conjunction with butterfly in order to get the needed flap travel.  BTW, This also would be a much lower priority that the telemetry bugs for instance for me, since obviously I have work-around for this.
[model06.zip](https://github.com/FrSkyRC/ETHOS-Feedback-Community/files/6571315/model06.zip)

**Comment by Liverboy77** [comment]
No flap position is top on switch -100

**Comment by Joetrin** [comment]
> @Joetrin same question for Aileron, Flap and Elevator (for Throttle I need to check myself)

Bonjour Bernard,

For for Crow, there's usually 2 aileron servos, and 2 flap servos and 2 elevator servos.. To do it in Mix would required only 1 offset or 2 offset positions per surface, and it's synchronized.  To do it on ouput as channel offset, it would need to be done in 6 places and it and you'd need to syncronize the changes when made.  (Note: Crow is when the outboard ailerons go upwards and flaps inboard go downward, they are used for air brakes, on Biplanes, Crow raises the ailerons on the top wings upwards and lowers the bottom ailerons downwards)

For Flaperons, there are 2 aileron servos per wing (total of 4 for both wings) and 2 elevator servos.  When either setting reflex or flaperon you'd need to do 1 setting offset setting for the aileron and 1 offset setting for elevator.  If it were done on output then again, 6 places to change.

Hope this helps.. Merci beaucoup.

**Comment by aalain8** [comment]
Hello,
I will try to explain differently what others are asking.
I like Ethos because the preset mixes are great, but for glider pilots some preset mixes are missing curves in each output to simplify the settings.

The F5J gliders have a very important subtrim on the flaps, between 60 and 80% of offset that I regulate in the exits.

when I adjust a camber mix on an inter, if you adjust the low position properly (towards yourself), it is impossible to raise the flaps as much as the ailerons.

The easiest way to get there is to duplicate the camber mix: One camber mix with ailerons and elevator and the other camber mix with flaps and an appropriate curve.

If you are a perfectionist and want to adjust the rudders to the mm, you would have to duplicate 4 camber mixes, one Ail.L / Ail.R / Vol.L / Vol.R / elevator.

Only the buterfly function worked perfectly on my 4 gliders.

On an airplane I also encountered the problem of "Liverboy77" the flaps only swing down and aileron and elevator on both sides.
Same, I duplicated the mix.

Of course, there are other possibilities to make the settings, but what are the predefined mixes for?

Thank you very much for the work

**Comment by gjvo** [comment]
Again, i already asked it a day or 10 ago and i got the same answer that weight and offset belong together.
Take a look at another brand, they use offset.
I agree with Hobby4Life.
Other brands have independent offsets and weight.
The way Ethos is working now needs a lot of working around a problem that is probably already solved by other brands, so Frsky can do that also.

**Comment by lthole** [comment]
I agree, mixes like Camber and Butterfly need to have Offset. For example, for Butterfly/Crow the ailerons must only go up, and the flaps only go down. Additionally glider pilots use the Thr stick for Crow, with zero Crow when the stick is fully up, and full Crow with the stick fully down. As mentioned above, there are many many situations where offset is needed, and having to use curves just complicates things.

**Comment by gjvo** [comment]
Another issue connected with this is that you have to use a curve.
I started out with a Function X>0 or X<0 but that was not fluent enough.
So i think, i take a 3 point curve.
But you cannot change the position on the X line, it is fixed on -100, 0, +100.
So to make it fluent i need a 6 or more point curve.
Is it possible to make it that you can choose the position the same like the Y line?
That would be a nice option, because my 20 year old Graupner already had that.

GJ

**Comment by Joetrin** [comment]
@bsongis-frsky is there any additional information needed at this time?

Thank You

**Comment by bsongis** [comment]
Will be fixed with the new "Offset" mix, as mentionned by #138

**Comment by bsongis-frsky** [comment]
Will be fixed with the new "Offset" mix, as mentionned by #138

**Comment by jasc76** [comment]
I don’t know
It could have been so much easier to just have a min/max in the trim mix… with offset you still need to lines to add smth so easy as a modifier input😞😞😞😞😞😞😞
